lte/gateway/python/magma/pipelined/openflow/flows.py

```diff
@@ -135,16 +135,31 @@ def add_flow(datapath, table, match, actions=None, instructions=None,
 
     if actions is None:
         actions = []
-    reset_scratch_reg_actions = [
-        parser.NXActionRegLoad2(dst=reg, value=REG_ZERO_VAL)
-        for reg in SCRATCH_REGS]
-    actions = actions + reset_scratch_reg_actions
+    # As 4G GTP tunnel, Register value is not used.
+    # if use for default flow, 4G traffic is dropping.
+    if goto_table != 1:
+        reset_scratch_reg_actions = [
+             parser.NXActionRegLoad2(dst=reg, value=REG_ZERO_VAL)
+             for reg in SCRATCH_REGS]
+        actions = actions + reset_scratch_reg_actions
 
     inst = __get_instructions_for_actions(ofproto, parser,
                                           actions, instructions)
+
+    # For 5G GTP tunnel, goto_table is used for downlink and uplink
+    if goto_table:
+        inst.append(parser.OFPInstructionGotoTable(goto_table))
+
     ryu_match = parser.OFPMatch(**match.ryu_match)
 
-    mod = parser.OFPFlowMod(datapath=datapath, priority=priority,
+    # cookie value is used for discard and forward data for GTP
+    if cookie > 0:
+        mod = parser.OFPFlowMod(datapath=datapath, priority=priority,
+                            match=ryu_match,table_id=table,
```

Comment:
why do you need to check for cookie value.

Reply:
As OAI code, cookie value is used for discard data gtp tunnel and forward data gtp tunnel functionality.